DeviceAdministration/Infrastructure.UnitTests/TestStubs/AlertHistoryBlobReaderStub.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.Repository;
+using Ploeh.AutoFixture;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.UnitTests.TestStubs
+{
+    public class AlertHistoryBlobReaderStub : IBlobStorageReader
+    {
+        private readonly Fixture fixture;
+        private readonly int _year;
+        private readonly int _month;
+        private readonly int _date;
+        private readonly string _value;
+
+        public AlertHistoryBlobReaderStub(int year, int month, int date, string value)
+        {
+            fixture = new Fixture();
+            _year = year;
+            _month = month;
+            _date = date;
+            _value = value;
+        }
+
+        public IEnumerator<BlobContents> GetEnumerator()
```

Comment:
This seems a lot of infrastructure code that's required to setup the right format to test the alerts repository. Instead of writing manual fixup code to create the records, create a simple .txt file as an embedded resource for the project and then load that up as a stream.